### PR TITLE
Webview based registry browser

### LINKED ISSUE #2417: Implement UI changes in the Registry Webview
- [x] Add line numbers in the devfile.yaml view
- [x] Copy the devfile.yaml
- [x] Add a badge (from material UI) with the count of the starter projects present in the registry stack
- [x] Show the filter for all the registries added in the Component Registries View
- [x] Show Tooltip (Material UI) when hover over the registry in the filter
- [x] Search box should work based on the filter selected
- [x] Check boxes should use material UI theme
- [x] Add icon for `Show View` in the Component Registries to open the webview for the registry viewer
- [x] Remove the children nodes of the registries added

Relates to the Epic issue #2392

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Creating webview based devfile registry view

Fixes #2417

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,1622 +1,1663 @@
 {
-  "name": "vscode-openshift-connector",
```
Comment (dgolovin) [inline package.json:2]: Please, return original tab size back. It is hard to see what was actually changed.

**Review by dgolovin — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (msivasubramaniaan) [reply package.json]: Done.

File: src/componentTypesView.ts [hunk 2]
```diff
@@ -4,52 +4,38 @@
  *-----------------------------------------------------------------------------------------------*/
```
Comment (dgolovin) [inline src/componentTypesView.ts:4]: There should be no dramatic changes in this file. There should be command to open webview based Registry View and that is it.

File: src/componentTypesView.ts [hunk 3]
```diff
@@ -4,52 +4,38 @@
...
 import { vsCommand, VsCommandError } from './vscommand';
-import { Platform } from './util/platform';
 import validator from 'validator';
+import { BaseTreeDataProvider } from './base/baseTreeDataProvider';
+import { WorkspaceEntry, WorkspaceFolderComponent } from './componentsView';
+import { Application } from 'express';
+import { Project } from './odo/project';
 
-type ComponentType = DevfileComponentType | StarterProject | Registry;
+type ComponentType = Project | Application | WorkspaceEntry | WorkspaceFolderComponent;
```
Comment (dgolovin) [inline src/componentTypesView.ts:27]: This is registry tree view. It has nothing to do with OpenShift cluster. Projects, Applications and etc. do not belong to this view.

File: src/openshift/component.ts [hunk 4]
```diff
@@ -90,6 +91,11 @@ export class Component extends OpenShiftItem {
         return !!ws;
     }
 
+    @vsCommand('openshift.component.open.devfileRegistry')
```
Comment (dgolovin) [inline src/openshift/component.ts:94]: It should be moved to componentTypesView.ts and called from context menu or inline group on Registry tree item.

**Review by dgolovin — CHANGES_REQUESTED** [review]

File: src/openshift/component.ts (hunk 4 above)
Comment (dgolovin) [reply src/openshift/component.ts:94]: Even better to keep it in registryViewLoader.ts.

File: package.json [hunk 5]
```diff
@@ -930,6 +939,11 @@
         "view": "openshiftComponentsView",
         "contents": "You have no OpenShift Components in your workspace yet.\n[New Component](command:openshift.component.createFromLocal)",
         "enablement": "openshift.component.explorer.init"
+      },
+      {
+        "view": "openshiftComponentTypesView",
+        "contents": "Default Devfile Regisitries are Here.\n[Open Default Devfile Registry](command:openshift.component.open.devfileRegistry)",
```
Comment (dgolovin) [inline package.json:945]: This does not fit Component Types View purpose. Welcome contents appears only when view has no items. It means when there is no registries listed with 'odo registry list' it supposed to show welcome content. The button should trigger 'Add registry command' not 'Open Default registry'.

File: package.json [hunk 6]
```diff
@@ -930,6 +939,11 @@
...
+        "enablement": "openshift.component.explorer.init"
```
Comment (dgolovin) [inline package.json:946]: This enablement condition is not correct. It should be specific to Component Types View, not Components view (which shows components in workspace).

**Comment by dgolovin** [comment]
Please see https://github.com/msivasubramaniaan/vscode-openshift-tools/pull/1 for details.

**Comment by mohitsuman** [comment]
@msivasubramaniaan As reviewed and discussed, please add the latest changes and rebase the code, so we do a quick review of the PR. I have added a checklist on #2417 , please verify if we have fixed all the tasks.

**Pull request merged**